packages/framework/tests/Test/Codeception/AdminSelectbox.php

```diff
@@ -42,10 +44,66 @@ public static function createByCss(ActorInterface $tester, string $cssSelector):
     }
 
     /**
-     * @param string $value
+     * @param string $optionValue
+     */
+    public function select(string $optionValue): void
+    {
+        $wrapperClass = $this->getWrapperClass();
+
+        $this->tester->clickByCss(sprintf('.%s .selectric', $wrapperClass));
+
+        $index = $this->getOptionIndexByValue($optionValue);
+
+        $this->tester->clickByCss(sprintf('.%s .selectric-scroll li[data-index="%d"]', $wrapperClass, $index));
+    }
+
+    /**
+     * @param string $optionValue
      */
-    public function select(string $value): void
+    public function assertSelected(string $optionValue): void
     {
-        $this->tester->selectOptionByCssAndValue($this->cssSelector, $value);
+        $wrapperClass = $this->getWrapperClass();
+
+        $label = $this->getOptionLabelByValue($optionValue);
+
+        $message = sprintf('Label for value "%s" not found in admin selectbox "%s".', $optionValue, $this->cssSelector);
+        Assert::assertNotEmpty($label, $message);
+
+        $this->tester->seeInCss($label, sprintf('.%s .selectric', $wrapperClass));
+    }
+
+    /**
+     * @return string
+     */
+    protected function getWrapperClass(): string
```

Comment:
> Could you maybe describe this in the commit message, please? Or in the method annotation?

The same request applies to similar methods in `AdminCheckbox` and `AdminRadiobutton`

Reply:
Done in 1a264bfd1 and 7ef16601c.

Reply:
It's not really because of more selectboxes on a page (the `$this->cssSelector` could handle that), it's more because the particular element is not easily targeted by a CSS selector.

When I'm trying to click a particular item in a menu, I need to target list item with a particular `data-index` attribute which resides in `.selectric-scroll` of a common parent wrapper element with the selectbox (that is defined by the `$this->cssSelector`).
I saw that this targeting was pretty complex, so I've marked the element in question with a generated class via JS to avoid all that complexity.

Reply:
ok, thanks for the clarification!